pkg/provision/provision.go

```diff
@@ -153,6 +153,12 @@ func copyHostCerts(authOptions auth.Options) error {
 		if err != nil {
 			return errors.Wrapf(err, "open cert file: %s", src)
 		}
+		defer func() {
```

Comment:
move the defer before 

```
if err != nil
```
here and other places

Reply:
Why do we need to do it? If `assets.NewFileAsset` returns an error, `f` is always nil, we can't close it

Reply:
you are right, in this case we always return nil file, but as a good practice wouldn't hurt to future proof it, in case the implementation changes and we return a file despite a specific error